README.md

```diff
@@ -146,24 +151,79 @@ CMSIS package will generate linker script for your device automatically (target
+[cmake/FindFreeRTOS](cmake/FindFreeRTOS.cmake) - finds FreeRTOS sources in location specified by
+`FREERTOS_PATH` (*default*: `/opt/FreeRTOS`) variable and format them as `IMPORTED` targets.
+`FREERTOS_PATH` can be either the path to the whole
+[FreeRTOS/FreeRTOS](https://github.com/FreeRTOS/FreeRTOS) github repo, or the path to
+FreeRTOS-Kernel (usually located in the subfolder `FreeRTOS` on a downloaded release).
 
-Typical usage:
+You can either use the FreeRTOS kernel provided in the Cube repositories, or a separate
+FreeRTOS kernel. The Cube repository also provides the CMSIS RTOS and CMSIS RTOS V2 implementations.
+If you like to use the CMSIS implementations, it is recommended to also use the FreeRTOS sources
+provided in the Cube repository because the CMSIS port might be incompatible to newer kernel
+versions.
+
+You can specify to use CMSIS with a `CMSIS` target and by finding the CMSIS `RTOS` package.
+To select which FreeRTOS to use, you can find the package for a specific FreeRTOS port and then
+link against that port target within a specific namespace.
+
+Typical usage for a H7 device when using the M7 core, using an external kernel without CMSIS
+support. The FreeRTOS namespace is set to `FreeRTOS` and the `ARM_CM7` port is used:
+
+```cmake
+find_package(CMSIS COMPONENTS STM32H743ZI STM32H7_M7 REQUIRED)
+find_package(FreeRTOS ARM_CM7 REQUIRED)
+target_link_libraries(${TARGET_NAME} PRIVATE
+    ...
+    FreeRTOS::ARM_CM7
+)
+```
+
+Typical usage for a F4 device, using an external kernel without CMSIS support.
+The FreeRTOS namespace is set to `FreeRTOS` and the `ARM_CM4F` port is used:
 
 ```cmake
 find_package(FreeRTOS COMPONENTS ARM_CM4F REQUIRED)
-target_link_libraries(... FreeRTOS::ARM_CM4F)
+target_link_libraries(${TARGET_NAME} PRIVATE
+    ...
+    FreeRTOS::ARM_CM4F
+)
 ```
 
-The following FreeRTOS ports are supported: `ARM_CM0`, `ARM_CM3`, `ARM_CM4F`, `ARM_CM7`.
+Another typical usage using the FreeRTOS provided in the Cube repository and the CMSIS support.
+The FreeRTOS namespace is set to `FreeRTOS::STM32::<FAMILY>` and the `ARM_CM7` port is used:
+
+```cmake
+find_package(CMSIS COMPONENTS STM32H743ZI STM32H7_M7 RTOS REQUIRED)
+find_package(FreeRTOS COMPONENTS ARM_CM7 STM32H7 REQUIRED)
+target_link_libraries(${TARGET_NAME} PRIVATE
+    ...
+    FreeRTOS::STM32::H7::M7::ARM_CM7
+    CMSIS::STM32::H7::M7::RTOS
+)
+```
+
+The following CMSIS targets are available in general:
+
+* `CMSIS::STM32::<Family>::RTOS`
+* `CMSIS::STM32::<Family>::RTOS_V2`
+
+The following additional FreeRTOS targets are available in general to use the FreeRTOS provided
+in the Cube repository
+
+* `FreeRTOS::STM32::<Family>`
```

Comment:
When a cube path is provided this should be populated and usable but it is not the case from my tests.
Maybe `FREERTOS_PATH` should not be mandatory when a `STM32_CUBE_<FAMILY>_PATH` is provided?

Reply:
Have you added the `STM32<Family>` to the FreeRTOS components? This is necessary to use the target inside the namespace I think

Reply:
No I have not. And it was not enough to solve the issue. 
But this did: 
```Cmake
#if(USE_CMSIS_RTOS OR USE_CMSIS_RTOS_V2)
        list(APPEND FREERTOS_COMP_LIST STM32H7)
        set(FREERTOS_H7_NAMESPACE ${FREERTOS_NAMESPACE}::STM32::H7::M7)
#else()
#    set(FREERTOS_H7_NAMESPACE ${FREERTOS_NAMESPACE})
#endif()
```
You are more expert that I am. Does the example need update ?

Reply:
Ah was that the example? That is weird, `FREERTOS_PATH` should not be necessary.. What was the build generation command?

Reply:
I just tested it. I loaded the STM32_CUBE_H7_PATH by setting it as an environment with
`export STM32_CUBE_H7_PATH=<pathToCube>` and the toolchain with `export STM32_TOOLCHAIN_PATH=<pathToToolchain>`

I the ran the following command inside `example/freertos/build`:

```cmake
cmake -DUSE_CMSIS_RTOS=ON -DFREERTOS_H743ZI_EXAMPLE=ON -DSTM32_TOOLCHAIN_PATH=$STM32_TOOLCHAIN_PATH ..
```

For CMSIS V2 I ran

```sh
cmake -DUSE_CMSIS_RTOS_V2=ON -DFREERTOS_H743ZI_EXAMPLE=ON -DSTM32_TOOLCHAIN_PATH=$STM32_TOOLCHAIN_PATH ..
```

Reply:
Ah, if neither `USE_CMSIS_RTOS` nor `USE_CMSIS_RTOS_V2` is specified the example could theoretically use both external kernel or STM32H7 kernel. But then what is taken as the default choice? Maybe introduce another option for that?

Reply:
Okay, I updated it. It should work for your case now as well. There is an option to specify whether the Cube FreeRTOS or an external kernel is picked.